### ISSUE TITLE
Encounter some problems when simulating wake steering control using FAST.Farm

### ISSUE THREAD

**Issue by VenusJinxin** [OpenFAST/openfast#2493]
Hi everyone,
Recently I'm learning to use FAST.Farm to simulate the wake steering control of wind farm. What I do firstly I use FLORISSE_M to solve a set of optimal yaw angle to maximze the total power generation of WF, and secondly I set static yaw angle values for each wind turbine in FAST.Farm to test the control performance. However, the result is not ideal. 
There are 3 WTs in my case, and arrange in a row along the wind direction and the distance is 5D. The results of FLORISSE_M are as follows:
![e7eaf453c8862f1e683dfa5ff2ba611](https://github.com/user-attachments/assets/f7fda89c-3c07-485c-91c1-9023a8a98cea)
![image](https://github.com/user-attachments/assets/93bc5611-3de3-4bb5-97fe-819e96ec3195)
However when I simulate it in FAST.Farm, the total power generation of WF is less than the case without yaw angle misalignment. The total power and power of each WT under the two methods are shown in the following figures:
![image](https://github.com/user-attachments/assets/1ee887b0-f921-4501-9bd5-58ade2afdc55)
![image](https://github.com/user-attachments/assets/08f908c1-a83c-4a0c-9bcb-2db30c21eff3)
![image](https://github.com/user-attachments/assets/2146c4a2-ce1a-43be-b0b4-6d13be3de17b)
![image](https://github.com/user-attachments/assets/c54c406f-8297-4584-9e49-212921dde43a)
The first figure is the total power and the rest 3 figures are power of WT1~3. The blue line represents the control without yaw. We can see that after the yaw control, the total power actually decreased.  
At the same time, I would like to ask a question. Each of my FAST file uses the default DISCON controller, and I  modify the parameters related to yaw in DISCON.IN and set YCMode=5 in ServoDyn, but the yaw of the wind turbine was not activated. I set the initial value of NacYaw in ElastoDyn based on the answers to other questions, and set YawNeut=NacYaw and YawManRat=0.2 in ServoDyn, and in this way, yaw misalignment has been achieved both in OpenFAST and FAST.Farm. 
The ideal results would be to increase the power of the entire WF through the yaw of the upstream WT, but this cannot be achieved in my case. Even if I set the yaw angle of WT1 to 10 ° or 5 ° and set the other WTs to 0 °, the total power still cannot be improved. I really want to know if there is anything wrong with my settings? Or is it due to inaccurate optimization results of yaw angle? 
 Yours Sincerly, 
 Venus

**Comment by jjonkman** [comment]
Dear @VenusJinxin,

Can you clarify:

- What wind turbine are you modeling, e.g., the NREL 5-MW baseline or the IEA Wind 15-MW RWT?
- What DISCON controller are you using?
- Which wake formulation (`Mod_Wake`) are you using in FAST.Farm?  FYI: The curled wake model (`Mod_Wake` = 3) is more accurate for wake-steering applications.
- What inflow conditions are you using in FAST.Farm?
- Are you using DEFAULT wake-dynamics parameters or did you calibrate them in some way?


Regarding yaw control, `YCMode` = 5 in ServoDyn is only need if you want active yaw control.  You can use `YCMode` = 0 with `YawNeut` = `NacYaw` in ServoDyn if you want fixed yaw when `YawDOF` is enabled in ElastoDyn.

Best regards,

**Comment by VenusJinxin** [comment]
Dear @jjonkman ,
Thanks for your detailed advice. I'm not quite sure why I made the modification to C_NearWake, maybe it was due to my mistake. 
Yesterday I downloaded a new r-test document and made modifications according to my previous simulation. I also conducted simulations under steady wind conditions. I found that when the wind speed is constant, the power increase caused by yaw is very significant; However, under turbulent wind conditions, it is not satisfactory. This may be because when the wind speed fluctuates too much, and the optimal yaw angle at different wind speeds is also inconsistent. 
Perhaps my problem comes from an inappropriate control scheme used, not FAST Farm :)
Best Regards, 
Venus
